tracer/src/Datadog.Trace/Processors/NormalizerTraceProcessor.cs

```diff
@@ -126,7 +126,7 @@ public Span Process(Span span)
                 string env = span.GetTag("env");
                 if (!string.IsNullOrEmpty(env))
                 {
-                    span.SetTag("env", TraceUtil.NormalizeTag(env));
+                    span.Tags.SetTag("env", TraceUtil.NormalizeTag(env));
```

Comment:
Sorry for the delay! Actually, it is because of the `IsFinished==true` property. I've set up the processors to run after the spans have been closed and when they're on their way to being serialized, so we can't use the public `ISpan.SetTag` interface here

Reply:
Ah yep thanks that makes sense!